pkg/controller.v2/informer.go

```diff
@@ -92,7 +93,8 @@ func tfJobFromUnstructured(obj interface{}) (*tfv1alpha2.TFJob, error) {
 	// This is a simple validation for TFJob to close
 	// https://github.com/kubeflow/tf-operator/issues/641
 	// TODO(gaocegege): Add more validation here.
-	if err != nil || tfjob.Spec.TFReplicaSpecs == nil {
+	validationErr := validation.ValidateAlpha2TFJobSpec(&tfjob.Spec)
+	if err != nil || tfjob.Spec.TFReplicaSpecs == nil || validationErr != nil {
```

Comment:
Please move the check `tfjob.Spec.TFReplicaSpecs == nil` to the function ValidateAlpha2TFJobSpec